code/src/MapAPI/HeatmapLayerManager.ts

```diff
@@ -79,6 +79,10 @@ namespace MapAPI.HeatmapLayerManager {
         configs: string
     ): OSFramework.HeatmapLayer.IHeatmapLayer {
         const map = GetMapByHeatmapLayerId(heatmapLayerId);
+        OSFramework.Helper.ValidateFeatureProvider(
+            map,
+            OSFramework.Enum.Feature.HeatmapLayer
+        );
         if (!map.hasHeatmapLayer(heatmapLayerId)) {
             const _heatmapLayer =
                 GoogleProvider.HeatmapLayer.HeatmapLayerFactory.MakeHeatmapLayer(
```

Comment:
What happens when this code is then ran?

Reply:
It was creating the provider behind the scenes but it was never associating it to the Heatmap Layer (os object) itself.
Fixed it!